numba/tests/test_tuples.py

```diff
@@ -361,6 +361,18 @@ def test_le(self):
 
 class TestNamedTuple(TestCase, MemoryLeakMixin):
 
+    def test_exception(self):
+        pyfunc = lambda: types.NamedTuple(types.uint32, 'p')
```

Comment:
I am not sure we need the `pyfunc` here. The term `pyfunc` usually refers to the python original of an JIT compiled function, your use of this variable here is somewhat unusual and I think you can just call `types.NamedTuple`. in the `with` statement below.